group-coordinator/src/main/java/org/apache/kafka/coordinator/group/Group.java

```diff
@@ -200,4 +200,11 @@ void validateOffsetFetch(
      * Requests a metadata refresh.
      */
     void requestMetadataRefresh();
+
+    /**
+     * Returns whether this group should be expired or not
+     *
+     * @return whether the group should be expired
+     */
+    boolean shouldExpire();
```

Comment:
You could have a default implementation that returns true so it's not necessary to put identical implementations in all of the subclasses which do not override.